episodes/quality-control-and-manual-segmentation.md

```diff
@@ -0,0 +1,547 @@
+different pixel value. We can find the maximum value via the console like so:
+
+```python
+# Get the instance segmentation data
+image = viewer.layers['instance_seg'].data
+
+# Print the maximum pixel value
+print(image.max())
+
+```
+
+```output
+19
+
+```
+
+Instance segmentations tend to have many more pixel values, as there is one per 
+object 'instance'. When working with images with very large numbers of cells, 
+you can quickly end up with hundreds or thousands of pixel values in an instance 
+segmentation.
+
+So to summarise - segmentations are just images where the pixel values represent 
+specific categories, or individual instances of that category. You will 
+sometimes see them referred to as 'label images', especially in the context of 
+Napari. Segmentations with only two values (e.g. 0 and 1 for background vs some 
+specific category) are often referred to as 'masks'.
+
+::::::::::::::::::::::::::::::::::::: challenge 
+
+## Segmentation shapes and data types
+
+Copy and paste the following into Napari's console:
+
+```python
+nuclei = viewer.layers["nuclei"].data
+semantic_seg = viewer.layers["semantic_seg"].data
+instance_seg = viewer.layers["instance_seg"].data
+```
+
+- What shape are the `nuclei`, `semantic_seg` and `instance_seg` images?
+
+- What pattern do you see in the shapes? Why?
+
+- What bit-depth do the `semantic_seg` and `instance_seg` images have?
+
+- If I want to make an instance segmentation of 100 nuclei - what's the 
+minimum bit depth I could use? (assume we're storing unsigned integers) Choose 
+from 8-bit, 16-bit or 32-bit.
+
+- If I want to make an instance segmentation of 1000 nuclei - what's the 
+minimum bit depth I could use? (assume we're storing unsigned integers) Choose 
+from 8-bit, 16-bit or 32-bit.
+
+Recall that we looked at image shapes and bit-depths in the ['What is an image?' 
+episode](what-is-an-image.md).
+
+:::::::::::::::::::::::: solution 
+
+### Solution
+
+### Shapes + pattern
+
+```python
+print(nuclei.shape)
+print(semantic_seg.shape)
+print(instance_seg.shape)
+```
+
+```output
+(60, 256, 256)
+(60, 256, 256)
+(60, 256, 256)
+```
+
+All the images have the same shape of (60, 256, 256). This makes sense as 
+'semantic_seg' and 'instance_seg' are segmentations made from the nuclei image. 
+Therefore, they need to be exactly the same size as the nuclei image in order to 
+assign a label to each pixel.
+
+### Bit depths
+
+```python
+print(semantic_seg.dtype)
+print(instance_seg.dtype)
+```
+
+```output
+int8
+int32
+```
+
+The data type (`.dtype`) of 'semantic_seg' contains the number 8, showing that 
+it is an 8-bit image. The data type of 'instance_seg' contains the number 32, 
+showing that it is a 32-bit image. Note that in this case, as the instance 
+segmentation only contains 19 nuclei, it could also have been stored as 8-bit 
+(and probably should have been, as this would provide a smaller file size!).
```

Comment:
This raises the question of why it wasn't stored as 8-bit and how you would go about converting it. Probably worth having an answer to hand for the lesson.